From e552ee5c77ab0d39de8f60055b5e1f4974982e10 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 25 Apr 2019 12:56:26 -0700
Subject: [PATCH] OS-7765 clarify that vminfod/client.js is not a public
 promised interface Reviewed by: Dave Eddy <dave.eddy@joyent.com> Approved by:
 Dave Eddy <dave.eddy@joyent.com>

---
 man/src/vminfo.1m.md                  | 4 ++--
 src/vm/node_modules/vminfod/client.js | 5 +++--
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/man/src/vminfo.1m.md b/man/src/vminfo.1m.md
index aca2ede5..684ccd29 100644
--- a/man/src/vminfo.1m.md
+++ b/man/src/vminfo.1m.md
@@ -47,5 +47,5 @@ determine service level health and status.
 
 This tool should be used for interactive output only, and is not meant to
 provide a stable interface to use for vminfod.  If you are trying to interface
-directly with `vminfod` use the `vminfod/client` JavaScript library or use the
-`vmadm(1M)` command that will use `vminfod` under the hood automatically.
+with `vminfod` use the `vmadm(1M)` command (especially `vmadm events`) and, for
+internal platform code, the `vminfod/client` Node.js library.
diff --git a/src/vm/node_modules/vminfod/client.js b/src/vm/node_modules/vminfod/client.js
index 23616817..aa889279 100644
--- a/src/vm/node_modules/vminfod/client.js
+++ b/src/vm/node_modules/vminfod/client.js
@@ -20,7 +20,7 @@
  *
  * CDDL HEADER END
  *
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  *
  */
 
@@ -28,7 +28,8 @@
  * - VminfodClient
  *
  * A client class that can be used to make basic GET requests to the vminfod
- * service.
+ * service. This interface is only intended for internal-to-the-platform
+ * usage.
  *
  * var vc = new require('vminfod/client').VminfodClient();
  *
-- 
2.21.0

